Mathlib/CategoryTheory/Filtered/OfColimitCommutesFiniteLimit.lean

```diff
@@ -0,0 +1,35 @@
+/-
+Copyright (c) 2024 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import Mathlib.CategoryTheory.Limits.Filtered
+import Mathlib.CategoryTheory.Limits.Yoneda
+
+/-!
+# If colimits of shape `K` commute with finite limits, then `K` is filtered.
+-/
+
+universe v u
+
+namespace CategoryTheory
+
+variable {K : Type v} [SmallCategory K]
+
+open Limits
+
+/-- A converse to `colimitLimitIsoLimitColimit`: if colimits of shape `K` commute with finite
+limits, then `K` is filtered. -/
+theorem isFiltered_of_nonempty_limit_colimit_to_colimit_limit
+    (h : ∀ {J : Type v} [SmallCategory J] [FinCategory J] (F : J ⥤ K ⥤ Type v),
+      Nonempty (limit (colimit F.flip) ⟶ colimit (limit F))) : IsFiltered K := by
+  refine IsFiltered.iff_nonempty_limit.2 (fun {J} _ _ F => ?_)
+  suffices Nonempty (limit (colimit (F.op ⋙ coyoneda).flip)) by
+    obtain ⟨X, y, -⟩ := Types.jointly_surjective' (this.map (h (F.op ⋙ coyoneda)).some).some
+    exact ⟨X, ⟨(limitObjIsoLimitCompEvaluation (F.op ⋙ coyoneda) _).hom y⟩⟩
+  let _ (j : Jᵒᵖ) : Unique ((colimit (F.op ⋙ coyoneda).flip).obj j) :=
+    ((colimitObjIsoColimitCompEvaluation (F.op ⋙ coyoneda).flip _
+      ≪≫ Coyoneda.colimitCoyonedaIso _)).toEquiv.unique
```

Comment:
```suggestion
    ((colimitObjIsoColimitCompEvaluation (F.op ⋙ coyoneda).flip _ ≪≫
      Coyoneda.colimitCoyonedaIso _)).toEquiv.unique
```

Reply:
Thanks, I'm not sure why this particular style rule is so hard for me to remember...